libkineto/src/CuptiActivityProfiler.cpp

```diff
@@ -1185,6 +1196,24 @@ const time_point<system_clock> CuptiActivityProfiler::performRunLoopStep(
         LOG(INFO) << "Tracing complete.";
         VLOG_IF(1, currentIter > 0) << "This state change was invoked by application's step() call";
 
+        if (currentIter > 0) {
```

Comment:
Please add a comment that for iteration based collection we want to stop trace iteration async in a callback thread.